File: general/simulation_stack/simulation_stack.java
//JAVA 
import java.util.*; 

@SuppressWarnings({"unchecked", "deprecation"})
public class Main { 
    public static void main(String[] args) { 
        Stack<Integer> s = new Stack<>();
        int[] rs = new int[10000];
        int k=0;
        Scanner sc = new Scanner(System.in);
        while(true){
            String cmd = sc.next();
            if (cmd.equals("#")){
                break;
            } else if (cmd.equals("PUSH")) {
                int curr = sc.nextInt();
                s.push(curr);
            } else if (cmd.equals("POP")) {
                if (!s.empty()){
                    int curr = s.peek();
                    rs[k] = curr;
                    s.pop();
                }
                k=k+1;
            }
        }
        for (int i=0;i<k;i++){
            if (rs[i] != 0){
                System.out.println(rs[i]);
            } else {
                System.out.println("NULL");
            }
        }
    }
}

File: general/simulation_stack/simulation_stack.cpp
//CPP 
#include <bits/stdc++.h>
using namespace std;
int main(){
	stack<int> s;
	int rs[10000];
	int k = 0;

	while(1){
        string cmd;
        cin >> cmd;
        if (cmd == "PUSH"){
            int curr;
            cin >> curr;
            s.push(curr);
        } else if (cmd == "POP"){
            if (!s.empty()){
                int curr = s.top();
                rs[k] = curr;
                s.pop();
            }
            k++;
        } else if (cmd == "#"){
            break;
        }
	}

	for (int i=0;i<k;i++){
        if (rs[i] != 0){
            cout << rs[i] << endl;
        } else {
            cout << "NULL" << endl;
        }
	}
    
	return 0;
}

